src/invoice.rs

```diff
@@ -133,16 +159,21 @@ impl std::fmt::Display for RgbInvoice {
             write!(f, "{amt}+")?;
         }
         write!(f, "{}", self.beneficiary)?;
-        if !self.unknown_query.is_empty() {
+        if self.expiry.is_some() || !self.unknown_query.is_empty() {
```

Comment:
It can be done it in a future, but for not forgetting to add conditions here (and in other places) we should introduce `fn has_params(&self) -> bool` method and use it here.

Reply:
of course. updated